File: solutions/cpp/log-levels/1/log_levels.cpp
#include <iostream>
#include <string>

namespace log_line {
std::string message(std::string line) {
    int ind = line.find(": ");
    return line.substr(ind + 2);
}

std::string log_level(std::string line) {
    int start = line.find("[");
    int end = line.find("]");
    
    return line.substr(start + 1, end - start - 1);
}

std::string reformat(std::string line) {
     return message(line) + " (" + log_level(line) + ")";
}
}  // namespace log_line

File: solutions/csharp/log-levels/1/LogLevels.cs
static class LogLine
{
    public static string Message(string logLine)
    {
        return logLine.Split(':')[1].Trim();
    }

    public static string LogLevel(string logLine)
    {
        string logLevel = logLine.Split(':')[0].Trim();
        logLevel = logLevel.Substring(1, logLevel.Length - 2);
        return logLevel.ToLower();
        
    }

    public static string Reformat(string logLine)
    {
        return $"{Message(logLine)} ({LogLevel(logLine)})";
    }
}
